grammars/javascript.cson

```diff
@@ -513,7 +513,16 @@
     'include': '#methods'
   }
   {
-    'match': '\\b(class)(?:(?:\\s+(extends)\\s+(\\w+(?:\\.\\w*)?))|(?:(?:\\s+(\\w+(?:\\.\\w*)?))(?:\\s+(extends)\\s+(\\w+(?:\\.\\w*)?))?))'
+    'match': '''(?x)
+      \\b(class)
+      (?:
+        (?: \\s+(extends) \\s+ ([a-zA-Z_$][\\w$]*) )
+        |
+        (?: (?: \\s+ ([a-zA-Z_$][\\w$]*) )
+          (?: \\s+ (extends) \\s+ ([a-zA-Z_$][\\w$]*) )?
+        )
+      )
+    '''
```

Comment:
We haven't decided whether or not to proceed with that one yet.  If we do, I'll ask you to :).

Also, can you remove the spaces in this regex?  I think they attract too much attention to themselves for some reason.